src/qutip_qip/circuit/circuit.py

```diff
@@ -1119,17 +997,19 @@ def draw(
                     Note : User should have write access to the location.
         """
 
-        if file_type == "svg":
-            mode = "w"
-        else:
-            mode = "wb"
-            if file_type == "png" and not dpi:
-                dpi = 100
-        image_data = self._raw_img(file_type, dpi)
-        with open(
-            os.path.join(file_path, file_name + "." + file_type), mode
-        ) as f:
-            f.write(image_data)
+        if renderer == "latex":
+            if file_type == "svg":
+                mode = "w"
+            else:
+                mode = "wb"
+                if file_type == "png" and not dpi:
+                    dpi = 100
+            latex = TeXRenderer(self)
+            image_data = latex.raw_img(file_type, dpi)
+            with open(
+                os.path.join(file_path, file_name + "." + file_type), mode
+            ) as f:
+                f.write(image_data)
```

Comment:
Please add a else statement here and an raise an error saying something like `f"Unknown {renderer}, please choose from "latex_render", ... and ...` etc. Right now there is only latex rendering of course.

Reply:
Added the error message and kept the options simple: latex, matplotlib, and text for easy recall.